csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -198,14 +254,17 @@ module DataFlowPrivateCached {
     ) {
       cfn.getElement() instanceof DelegateArgumentToLibraryCallable and
       any(DelegateArgumentConfiguration x).hasExprPath(_, cfn, _, call)
-    }
+    } or
+    TMallocNode(ControlFlow::Nodes::ElementNode cfn) { cfn.getElement() instanceof ObjectCreation }
```

Comment:
I still don't fully understand how this will help field flow, but I see it's a port from Java (https://github.com/Semmle/ql/blob/a8be7f2434dbc60f934d88d61524b93c1839f46f/java/ql/src/semmle/code/java/dataflow/internal/DataFlowUtil.qll#L21) so LGTM.

Reply:
I realize now that it probably doesn't make too much sense on this PR, but should hopefully be more clear when something similar to Java's https://github.com/Semmle/ql/blob/a8be7f2434dbc60f934d88d61524b93c1839f46f/java/ql/src/semmle/code/java/dataflow/internal/DataFlowUtil.qll#L234 is added.